telos-backend/src/routes/api/journal.js

```diff
@@ -17,10 +122,40 @@ router.get('/:date', (req, res) => {
 // 'date': date to insert widget into
 // 'position': position of widget in journal for the specified date
 // 'widget': type of widget i.e. 'calendar', etc.
-router.post('/', (req, res) => {
-  res.json({
-    endpoint: '/journal',
-    request: `POST date: ${req.body.date}, position: ${req.body.position}, widget: ${req.body.widget}`,
+router.post('/', async (req, res) => {
+  const date = req.body.date
+  const newWidget = new Widget({
+    date: date,
+    position: {
+      row: req.body.position.row,
+      col: req.body.position.col
+    },
+    type: req.body.type,
+  });
+  newWidget.save(async (err) => {
+    if (err) {
+      return res.status(400).json({ error: err });
+    }
+    let data;
+    const type = req.body.type;
+    switch(type){
+      case 'calendar':
+        data = await getCalendarDataForDay(date);
+        break
+      case 'todo':
+        data = await getTodosForDay(date);
+        break
+      case 'habit_tracker':
+        data = await getHabitsForDay(date);
+        break
+      case 'text':
```

Comment:
When a text widget is first made, there wont be any text data with that widgetId yet, so should we instead of finding, just create a new text data document? basically doing whats done in /text API POST action? In that case, we probably don't even need the text post method.